core/chains/evm/gas/arbitrum_estimator_test.go

```diff
@@ -0,0 +1,223 @@
+package gas_test
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/gas"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/gas/mocks"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+func TestArbitrumEstimator(t *testing.T) {
+	t.Parallel()
+
+	maxGasPrice := big.NewInt(100)
+	const maxGasLimit uint32 = 500_000
+	calldata := []byte{0x00, 0x00, 0x01, 0x02, 0x03}
+	const gasLimit uint32 = 80000
+
+	t.Run("calling GetLegacyGas on unstarted estimator returns error", func(t *testing.T) {
+		config := mocks.NewConfig(t)
+		rpcClient := mocks.NewRPCClient(t)
+		ethClient := mocks.NewETHClient(t)
+		o := gas.NewArbitrumEstimator(logger.TestLogger(t), config, rpcClient, ethClient)
+		_, _, err := o.GetLegacyGas(calldata, gasLimit, maxGasPrice)
+		assert.EqualError(t, err, "estimator is not started")
+	})
+
+	var zeros bytes.Buffer
+	zeros.Write(common.BigToHash(big.NewInt(0)).Bytes())
+	zeros.Write(common.BigToHash(big.NewInt(0)).Bytes())
+	zeros.Write(common.BigToHash(big.NewInt(123455)).Bytes())
+	t.Run("calling GetLegacyGas on started estimator returns estimates", func(t *testing.T) {
+		config := mocks.NewConfig(t)
+		config.On("EvmGasLimitMax").Return(maxGasLimit)
+		rpcClient := mocks.NewRPCClient(t)
+		ethClient := mocks.NewETHClient(t)
+		rpcClient.On("CallContext", mock.Anything, mock.Anything, "eth_gasPrice").Return(nil).Run(func(args mock.Arguments) {
+			res := args.Get(1).(*hexutil.Big)
+			(*big.Int)(res).SetInt64(42)
+		})
+		ethClient.On("CallContract", mock.Anything, mock.IsType(ethereum.CallMsg{}), mock.IsType(&big.Int{})).Run(func(args mock.Arguments) {
+			callMsg := args.Get(1).(ethereum.CallMsg)
+			blockNumber := args.Get(2).(*big.Int)
+			assert.Equal(t, gas.ArbGasInfoAddress, callMsg.To.String())
+			assert.Equal(t, gas.ArbGasInfo_getPricesInArbGas, fmt.Sprintf("%x", callMsg.Data))
+			assert.Equal(t, big.NewInt(-1), blockNumber)
+		}).Return(zeros.Bytes(), nil)
+
+		o := gas.NewArbitrumEstimator(logger.TestLogger(t), config, rpcClient, ethClient)
+		require.NoError(t, o.Start(testutils.Context(t)))
+		t.Cleanup(func() { assert.NoError(t, o.Close()) })
+		gasPrice, chainSpecificGasLimit, err := o.GetLegacyGas(calldata, gasLimit, maxGasPrice)
+		require.NoError(t, err)
+		assert.Equal(t, big.NewInt(42), gasPrice)
+		assert.Equal(t, gasLimit, chainSpecificGasLimit)
+	})
+
+	t.Run("gas price is lower than user specified max gas price", func(t *testing.T) {
```

Comment:
Description of this test is opposite of behavior.
Should be:
gas price is highler than user specified ...

Reply:
Or if this is indicating what correct behavior should be, then maybe it should say:
gas price should be lower than user ...

Reply:
Yeah I copy-pasted this error from the l2suggested tests. Will circle back with some of the other feedback in https://app.shortcut.com/chainlinklabs/story/52435